task5/src/routes/group/index.js

```diff
@@ -1,30 +1,32 @@
 import express from 'express';
 import GroupService from '../../services/groupService';
 import db from '../../models';
+import { apiLoggerMiddleware as loggerMiddleware } from '../../utils/logger';
+import { ErrorHandler, populateErrorWithHadledFields } from '../../utils/errorHandler';
 
 const router = express.Router();
 const groupService = new GroupService(db);
 
 router
     .route('/:id')
-    .get(async (req, res) => {
+    .get(loggerMiddleware, async (req, res, next) => {
         try {
             const { id: groupId } = req.params;
             const group = await groupService.getById(groupId);
             if (!group) {
-                res.status(404).json({
-                    message: 'No group found'
-                });
+                throw new ErrorHandler(
+                    404,
+                    `Group with id ${groupId} not found`
+                );
             } else {
                 res.json(group);
             }
+            return next();
         } catch (err) {
-            res.status(err.status || 500).json({
-                message: err
-            });
+            return next(populateErrorWithHadledFields(err, req));
```

Comment:
Is it OK to modify err object to make sure we can retrieve some properties in error handler middleware?